### PR TITLE
fix(platform): prevent filter input from losing focus when entering s…

### LINKED ISSUE #12285: [P4] [Sourcing] [Platform] Table column filtering does not allow multiple words
#### Is this a bug, enhancement, or feature request?
Bug in Angular 15

#### Describe your proposal.
The column filtering does not allow multiple words. When users attempt to enter spaces, the filter textbox loses focus and the space character is not entered. This can be a problem, since users may need to include spaces in their filter query.

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
Angular 15

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
Video attached. In this case, the attempted query is "Stocked on demand", but users can only get so far as to type "Stocked". Attempting to enter the space character makes it lose focus.

https://github.com/user-attachments/assets/520f1eec-8950-40e4-93d3-8f85fc027ab5

#### Did you check the documentation and the API?
Yes

#### Did you search for similar issues?
Yes

#### Is there anything else we should know?

**_IMPORTANT: Please refrain from providing links or screenshots of SAP's internal information, as this project is open-source, and its contents are accessible to anyone._**

### PR DESCRIPTION
**Pull request by dpavlenishvili**
…paces

## Related Issue(s)

closes #12285 

### Before:
image

### After:
image

### CODE REVIEW

File: libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.html [hunk 1]
```diff
@@ -99,6 +99,7 @@
                     fd-form-control
                     [id]="'fdp-table-column-filtering-' + column.name"
                     (keydown.enter)="_setColumnHeaderFilterBy(column.key, $any($event.target)?.value)"
+                    (keydown)="$event.stopPropagation()"
```
Comment (mikerodonnell89) [inline libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.html:102]: I am confused why this works but it does

Comment (dpavlenishvili) [reply libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.html:102]: The second (keydown) event listener with stopPropagation() ensures that the keydown event does not bubble up the DOM tree. This prevents unintended behavior such as triggering other event listeners higher in the component hierarchy that could interfere with the expected behavior of input field. Without this, the keydown event might propagate to other components, such as the popover or list items, causing the input field to lose focus or trigger other actions.

Comment (mikerodonnell89) [reply libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.html:102]: I think leave a comment here explaining the situation, I'd be confused if I stumbled upon this without contrext

Comment (dpavlenishvili) [reply libs/platform/table/components/table-cell-header-popover/table-cell-header-popover.component.html:102]: image

**Comment by droshev** [comment]
@dpavlenishvili this change needs also to be donwported to ng-15. You can ask @mikerodonnell89 to downport it in a separate PR, thanks.

**Review by khotcholava — APPROVED** [review]

**Pull request merged**